### DISCUSSION TITLE
Why chromium as a rendering engine :)

### DISCUSSION THREAD

**Discussion by demobiel** [gotenberg/gotenberg#954] [Q&A]
Starting from: https://github.com/gotenberg/gotenberg/issues/952

We had some small issues with chromium, so we decided to try weasyprint which at the moment gives really good results. Expect that we now have to wrap a HTTP server around it. and I feel like we are actually redeveloping gotenberg, but just with another engine :)

Before we start doing that I am just wondering if you guys considered weasyprint as a rendering engine, or why you chose chromium?

Kind regards

Andries

**Comment by gulien** [comment]
Hello @demobiel,

There is actually an issue about that: https://github.com/gotenberg/gotenberg/issues/789.
TL;DR: it requires time I don't have 😄

**Reply by demobiel** [reply]
so now the question is: do I invest time in developing a wrapper, or do I invest time learning the go language and add is a module... I think the second option is the most honest towards the open source community.

If I would start this, could I ask you for occasional help if needed?

Where my first question is: where to start? :)

**Reply by gulien** [reply]
I mean, you're very welcome to try to add a new module to Gotenberg, and I'm ready to help if necessary 👍 

Best option might be to create a standalone module with the [template repository](https://github.com/gotenberg/gotenberg-template-repository).

You may then take a look at the [Chromium module](https://github.com/gotenberg/gotenberg/tree/main/pkg/modules/chromium) for a more advanced example. Other modules may also be a source of inspiration, for instance [PDFtk](https://github.com/gotenberg/gotenberg/tree/main/pkg/modules/pdftk) for CLI usage.

But first thing first, I guess [A Tour of Go](https://go.dev/tour/welcome/1) is a great introduction to the language.
